test/device_info_test.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:amplitude_flutter/src/constants.dart';
+import 'package:amplitude_flutter/src/device_info.dart';
+import 'package:device_info_platform_interface/device_info_platform_interface.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+class MockDeviceInfoPlatform
+    with MockPlatformInterfaceMixin
+    implements DeviceInfoPlatform {
+  @override
+  Future<AndroidDeviceInfo> androidInfo() async =>
+      AndroidDeviceInfo.fromMap(<String, dynamic>{'model': 'android-model'});
+
+  @override
+  Future<IosDeviceInfo> iosInfo() async =>
+      IosDeviceInfo.fromMap(<String, dynamic>{
+        'model': 'ios-model',
+      });
+}
+
+void main() {
+  DeviceInfo deviceInfo;
+
+  setUp(() {
+    SharedPreferences.setMockInitialValues({});
+    DeviceInfoPlatform.instance = MockDeviceInfoPlatform();
+    deviceInfo = DeviceInfo(false);
+  });
+
+  group('#regenerateDeviceId', () {
+    test('when platform is Android', () async {
```

Comment:
I tried to test when the platform is ios, but I couldn't mock this [information](https://github.com/nubank/Amplitude-Flutter/blob/d751d67392e9874d6b7d39d772167b0bdb2a31b7/lib/src/device_info.dart#L136) in a simple way, if you have any suggestions I'm open to listening.
 I didn't want to change the class and make it more testable because I'm trying to modify as little as possible

Reply:
I agree with out: you won't be able to mock this information without modifying the class. 😞